lib/apply-diff.ts

```diff
@@ -0,0 +1,121 @@
+		);
+	});
+
+	// Deleting Vorgänge in database
+	if (remoteVorgängeToDelete.length === 0) {
+		console.log("No Vorgänge to delete in database...");
+	} else {
+		console.log(`Deleting Vorgänge in database...`);
+		const progressBarDelete = new progress(":bar :current/:total", {
+			total: remoteVorgängeToDelete.length,
+		});
+		for (const vorgang of remoteVorgängeToDelete) {
+			await sql`delete from vorgang where vnr = ${vorgang.vnr}`;
+			progressBarDelete.tick();
+		}
+		progressBarDelete.terminate();
+	}
+
+	// Adding Vorgänge to database
+	if (localVorgängeToAddToRemote.length === 0) {
+		console.log("No Vorgänge to add to database...");
+	} else {
+		console.log(`Adding local Vorgänge to database...`);
+		const progressBarInsert = new progress(":bar :current/:total", {
+			total: localVorgängeToAddToRemote.length,
+		});
+		for (const vorgang of localVorgängeToAddToRemote) {
+			const vorgangResult = await sql`
+					INSERT INTO vorgang (VNr, VFunktion, ReihNr, VTyp, VTypL, VSys, VSysL, VIR, export_id)
+					VALUES (${checkIfArray(vorgang.VNr)}, ${checkIfArray(
+						vorgang.VFunktion,
+					)}, ${checkIfArray(vorgang.ReihNr)}, ${checkIfArray(
+						vorgang.VTyp,
+					)}, ${checkIfArray(vorgang.VTypL)}, ${checkIfArray(
+						vorgang.VSys,
+					)}, ${checkIfArray(vorgang.VSysL)}, ${checkIfArray(vorgang.VIR)}, ${
+						dbExport[0].id
+					})
+				RETURNING id
+			  `;
+
+			if (vorgang.Nebeneintrag) {
+				for (const nebeneintrag of vorgang.Nebeneintrag) {
+					await sql`
+						INSERT INTO nebeneintrag (ReihNr, Desk, vorgang_id)
+						VALUES (${checkIfArray(nebeneintrag.ReihNr)}, ${checkIfArray(
+							nebeneintrag.Desk,
+						)}, ${vorgangResult[0].id})
+					`;
+				}
+			}
+			if (vorgang.Dokument) {
+				for (const dokument of vorgang.Dokument) {
+					await sql`
+						INSERT INTO dokument (ReihNr, DHerk, DHerkL, Wp, DokArt, DokArtL, DokTyp, DokTypL, NrInTyp, Desk, Titel, DokNr, DokDat, LokURL, Sb, VkDat, HNr, Jg, Abstract, Urheber, vorgang_id)
+						VALUES (${checkIfArray(dokument.ReihNr)}, ${checkIfArray(
+							dokument.DHerk,
+						)}, ${checkIfArray(dokument.DHerkL)}, ${checkIfArray(
+							dokument.Wp,
+						)}, ${checkIfArray(dokument.DokArt)}, ${checkIfArray(
+							dokument.DokArtL,
+						)}, ${checkIfArray(dokument.DokTyp)}, ${checkIfArray(
+							dokument.DokTypL,
+						)}, ${checkIfArray(dokument.NrInTyp)}, ${checkIfArray(
+							dokument.Desk,
+						)}, ${checkIfArray(dokument.Titel)}, ${checkIfArray(
+							dokument.DokNr,
+						)}, ${checkIfArray(dokument.DokDat)}, ${checkIfArray(
+							dokument.LokURL,
+						)}, ${checkIfArray(dokument.Sb)}, ${checkIfArray(
+							dokument.VkDat,
+						)}, ${checkIfArray(dokument.HNr)}, ${checkIfArray(
+							dokument.Jg,
+						)}, ${checkIfArray(dokument.Abstract)}, ${checkIfArray(
+							dokument.Urheber,
+						)}, ${vorgangResult[0].id})
+					`;
+				}
+			}
+			progressBarInsert.tick();
+		}
+		progressBarInsert.terminate();
+	}
+
```

Comment:
You need to call `sql.end` if not there might be a dangling connection to the database.